porting/debug-build/udev.rst

```diff
@@ -1,16 +1,24 @@
 Add udev rules
 ==============
 
-Now that you're logged in, you must create some udev rules to allow some tests to access your hardware. Run the following command, replacing [codename] with your device's codename.::
+Now that you're logged in on the device, you must create some udev rules to allow some tests to access your hardware. Run the following command, replacing [codename] with your device's codename::
 
-    cat /var/lib/lxc/android/rootfs/ueventd*.rc|grep ^/dev|sed -e 's/^\/dev\///'|awk '{printf "ACTION==\"add\", KERNEL==\"%s\", OWNER=\"%s\", GROUP=\"%s\", MODE=\"%s\"\n",$1,$3,$4,$2}' | sed -e 's/\r//' >/etc/udev/rules.d/70-[codename].rules
+    cat /var/lib/lxc/android/rootfs/ueventd*.rc | grep ^/dev | sed -e 's/^\/dev\///' | awk '{printf "ACTION==\"add\", KERNEL==\"%s\", OWNER=\"%s\", GROUP=\"%s\", MODE=\"%s\"\n",$1,$3,$4,$2}' | sed -e 's/\r//' >/etc/udev/rules.d/70-[codename].rules
 
 Now, reboot the device.
 
-.. todo::
+Alternatively, if you want to create the file with the udev rules on your "host" machine you can run the following command inside your halium folder, replacing [codename] with your device's codename::
 
-    Update these instructions to update udev rules on "host" machine. The ueventd*rc files are already in $OUT/root/ , so it needs updating for path and then instead of writing it to /etc/udev/rules.d/70-codename.rules, writing it to local file and then adb pushing it while in recovery.
+    cat /out/target/product/[codename]/root/ueventd*.rc | grep ^/dev | sed -e 's/^\/dev\///' | awk '{printf "ACTION==\"add\", KERNEL==\"%s\", OWNER=\"%s\", GROUP=\"%s\", MODE=\"%s\"\n",$1,$3,$4,$2}' | sed -e 's/\r//' > 70-[codename].rules
 
-.. todo::
+You can then use either SSH or 'adb push' to push the file to the right location:
```

Comment:
No, that's the wrong character, check other pages for how to mark up inline code. Should be ``ssh`` or ``adb push``